### PR TITLE
chore(secrets): Add `OVERVIEW.md` for context on Zowe CLI and Zowe Explorer

### PR DESCRIPTION
**Pull request by traeok**
**What It Does**

This PR adds an "Overview" document to the root of the Secrets SDK package for more context on how it affects Zowe CLI and Zowe Explorer (both end users and extenders).

I don't feel that a version bump is necessary for this PR - it will be linked from Zowe Docs, and there are no other changes for the next patch version.

**Review Checklist**
I certify that I have:
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: packages/secrets/OVERVIEW.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+## Impact on End Users
+
+### Zowe CLI
+
+CLI users should not be affected by this change. It is intended to be a drop-in replacement for node-keytar, so existing credentials will still be accessible.
+
+### Zowe Explorer
+
+- **Most** Zowe Explorer users should not be affected by this change.
+- Some Zowe Explorer users may be affected by this change; specifically, users connecting over remote environments with Zowe Explorer might have to re-enter credentials.
```
Comment (anaxceron) [inline packages/secrets/OVERVIEW.md:20]: I'd avoid the semi-colon. Also, suggesting a minor edit (shown in bold): 
Some Zowe Explorer users may be affected by this change. Specifically, users connecting over remote environments with Zowe Explorer **might be required** to re-enter credentials.

File: packages/secrets/OVERVIEW.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+### Zowe CLI
+
+CLI users should not be affected by this change. It is intended to be a drop-in replacement for node-keytar, so existing credentials will still be accessible.
+
+### Zowe Explorer
+
+- **Most** Zowe Explorer users should not be affected by this change.
+- Some Zowe Explorer users may be affected by this change; specifically, users connecting over remote environments with Zowe Explorer might have to re-enter credentials. 
+    - Since keytar has been removed from VS Code for the next release, Zowe Explorer cannot access the local credential vault while the extension is running through a remote server. Those users will only have to re-enter their credentials once.
```
Comment (anaxceron) [inline packages/secrets/OVERVIEW.md:21]: Suggesting a minor edit here, which is to define "those users": Users who run Zowe Explorer through a remote server need to re-enter their credentials only once. 
(Idk if my facts are correct, but hopefully you can see where I'm going w/ this.)

File: packages/secrets/OVERVIEW.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+- Some Zowe Explorer users may be affected by this change; specifically, users connecting over remote environments with Zowe Explorer might have to re-enter credentials. 
+    - Since keytar has been removed from VS Code for the next release, Zowe Explorer cannot access the local credential vault while the extension is running through a remote server. Those users will only have to re-enter their credentials once. 
+    - "Remote environments" include:
+        - Remote SSH
+        - Remote Tunnel
+        - Remote Docker
+
+## Impact on Extenders
+
+**CAUTION:** Regardless of whether you extend Zowe CLI or Zowe Explorer, developers that directly reference the dependency `node-keytar` in their code need to use the new `keyring` module in `@zowe/secrets-for-zowe-sdk`.
```
Comment (anaxceron) [inline packages/secrets/OVERVIEW.md:29]: Up to you, but I'd make this a little stronger to convey the urgency/importance: "... in their code must use the new `keyring` moduel ..."

File: packages/secrets/OVERVIEW.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+### Zowe CLI
+
+- Developers can continue using the credential manager utilities from Imperative.
+- Developers that indirectly use Zowe CLI to store and load secure credentials will not be affected.
```
Comment (anaxceron) [inline packages/secrets/OVERVIEW.md:34]: replace "will" with "are" -- b/c these are things happening _now_

File: packages/secrets/OVERVIEW.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+### Zowe Explorer
+
+- Extenders that leverage Zowe Explorer API and Webpack (or another bundler) will have to update their extensions to provide prebuilds for the Secrets SDK. 
+    - This involves providing a folder named `prebuilds` with the Secrets SDK binaries alongside their extension root directory (same level as `package.json`).
+    - [Click here](https://github.com/zowe/zowe-cli/blob/master/packages/secrets/src/keyring/EXTENDERS.md#webpackingbundling-alongside-your-project) for more information on this process.
+- Extenders that do not use a bundler can continue using the credential manager utilities from Zowe Explorer API. These extensions will not be affected by this change.
```
Comment (anaxceron) [inline packages/secrets/OVERVIEW.md:41]: replace "will" with "are"

File: packages/secrets/OVERVIEW.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# Secrets SDK
+
+Secrets for Zowe SDK is a new development package used in the following Zowe releases:
```
Comment (anaxceron) [inline packages/secrets/OVERVIEW.md:3]: Should we also list Zowe 2.11 as its own item?

**Review by anaxceron — APPROVED** [review]
Thanks, Trae, I think this will be very helpful to all the users who read our new banner and are looking for more context.
Left a few minor comments.

**Review by KevinLoesch1 — APPROVED** [review]

File: packages/secrets/OVERVIEW.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+- Zowe CLI, v7.18.0 and above
+- Zowe Explorer and Zowe Explorer API
+    - v2.10.0 and above (v2 LTS)
+    - v1.22.5 and above (v1)
+- Zowe v2.11 bundle
```
Comment (t1m0thyj) [inline packages/secrets/OVERVIEW.md:8]: Should we be more specific about what the bundle is and where it comes from?

Maybe we could have 2 bullet points for Zowe CLI above:
> - Zowe CLI and SDKs
>     - v7.18.0 and above from NPM package
>     - v2.11.0 and above from bundle on zowe.org

Comment (traeok) [reply packages/secrets/OVERVIEW.md:8]: Going to omit "and SDKs", but I like this idea. Implemented in upcoming commit

File: packages/secrets/OVERVIEW.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        - Remote Docker
+
+## Impact on Extenders
+
+**CAUTION:** Regardless of whether you extend Zowe CLI or Zowe Explorer, developers that directly reference the dependency `node-keytar` in their code must use the new `keyring` module in `@zowe/secrets-for-zowe-sdk`.
+
+### Zowe CLI
+
+- Developers can continue using the credential manager utilities from Imperative.
+- Developers that indirectly use Zowe CLI to store and load secure credentials are not affected.
```
[the remark this answers is not in the document]
Comment (traeok) [reply packages/secrets/OVERVIEW.md:35]: Going to update the "caution" message at the top of the "Impact on Extenders" section:

```
**CAUTION:** Regardless of whether you extend Zowe CLI or Zowe Explorer, developers that directly reference `node-keytar` as a dependency or dev dependency must use the new `keyring` module in `@zowe/secrets-for-zowe-sdk`.
```

**Review by zFernand0 — APPROVED** [review]

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @traeok!

**Review by awharn — APPROVED** [review]

**Review by gejohnston — APPROVED** [review]
Looks good. It explains all of the situations.

**Review by adam-wolfe — APPROVED** [review]

**Pull request merged**